src/plugins/_test_utils.ts

```diff
@@ -0,0 +1,50 @@
+import { getLogger } from "std/log/mod.ts";
+import { Plugin, PluginDescriptor } from "./plugin.ts";
+import { _internals, ManagedPlugin, PluginState } from "./plugin_manager.ts";
+import { mergeUserConfigAndValidate } from "../config.ts";
+import { PluginManager } from "./mod.ts";
+import { stub } from "std/testing/mock.ts";
+
+export class TestPlugin extends Plugin {
+  get descriptor(): PluginDescriptor {
+    throw new Error("Method not implemented.");
+  }
+}
+
+export const testPluginMap = { "test-plugin": TestPlugin };
+
+export function testConfig() {
+  return mergeUserConfigAndValidate({
+    plugins: [{ enabled: true, id: "test-plugin" }],
+  });
+}
+
+export function mkPluginManager(
+  withPlugins: ManagedPlugin[] = [],
+  config = testConfig(),
+  pluginMap = testPluginMap,
+) {
+  const pluginsStub = withPlugins.length
+    ? stub(_internals, "plugins", () => withPlugins)
+    : null;
+
+  const cleanup = () => {
```

Comment:
when typescript 5.2 hits it would be pretty nice to remove this and use explicit resource management to clean up automatically